components/director/internal/destinationfetchersvc/service.go

```diff
@@ -99,57 +101,61 @@ func (d *DestinationService) generateClientBySubdomainLabel(ctx context.Context,
 
 	subdomain, ok := subdomainLabel.Value.(string)
 	if !ok {
-		log.C(ctx).Errorf("cannot cast label value as a string")
 		return nil, errors.New("cannot cast label value as a string")
 	}
 
 	region, ok := regionLabel.Value.(string)
 	if !ok {
-		log.C(ctx).Errorf("cannot cast label value as a string")
 		return nil, errors.New("cannot cast label value as a string")
 	}
 
 	instanceConfig, ok := d.DestinationsConfig.RegionToInstanceConfig[region]
 	if !ok {
-		log.C(ctx).Errorf("No destination instance credentials found for region '%s'", region)
-		return nil, errors.New(fmt.Sprintf("No destination instance credentials found for region '%s'", region))
+		return nil, fmt.Errorf("no destination instance credentials found for region '%s'", region)
 	}
 
 	client, err := NewClient(instanceConfig, d.APIConfig, subdomain)
 	if err != nil {
-		log.C(ctx).WithError(err).Error("Failed to create Destination API client")
-		return nil, err
+		return nil, errors.Wrap(err, "failed to create Destination API client")
 	}
 
 	return client, nil
 }
 
 // SyncTenantDestinations syncs destinations for a given tenant
 func (d *DestinationService) SyncTenantDestinations(ctx context.Context, tenantID string) error {
+	log.C(ctx).Infof("Starting sync of destinations for tenant '%s'", tenantID)
+
 	subdomainLabel, err := d.getSubscribedSubdomainLabel(ctx, tenantID)
 	if err != nil {
-		return err
+		return errors.Wrapf(err, "failed to get subdomain label for tenant '%s'", tenantID)
 	}
 
 	client, err := d.generateClientBySubdomainLabel(ctx, subdomainLabel)
 	if err != nil {
-		return err
+		return errors.Wrapf(err, "failed to create destinations client for tenant '%s'", tenantID)
 	}
 
+	// Make it debugf
+	log.C(ctx).Infof("Syncing destinations for tenant '%s' with subdomain '%s'", tenantID, subdomainLabel.Value)
```

Comment:
This log was for debug purposes only to track the execution time of the db requests